src/AutoCompleteTextField.js

```diff
@@ -297,7 +297,8 @@ class AutocompleteTextField extends React.Component {
 
   handleKeyDown(event) {
     const { helperVisible, options, selection } = this.state;
-    const { onKeyDown, passThroughEnter } = this.props;
+    const { onKeyDown, passThroughEnter, maxOptions } = this.props;
+    const optionsCount = maxOptions === 0 || maxOptions > options.length ? options.length : maxOptions
```

Comment:
nit: `;` in the end for consistency
Also a bit cleaner to my taste would be 
`const optionsCount = maxOptions > 0 ? Math.min(options.length, maxOptions) : options.length;`
It also covers case whenever folks passing `maxOptions = -1` expecting it to be infinity.